src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
```

Comment:
I think that we can also use `some` array method here, it may be more accurate having the prompt in mind, other than that we can also create function for checking if a mother has children 
```suggestion
  function hasChildren(person, people) {
      return people.some(otherPerson => otherPerson.mother === person.name);
}
```